src/Nethermind/Nethermind.Core/Crypto/Keccak.cs

```diff
@@ -85,12 +84,12 @@ public ValueKeccak(ReadOnlySpan<byte> bytes)
         {
             if (bytes.Length == 0)
             {
-                Bytes = OfAnEmptyString.Bytes;
+                _bytes = OfAnEmptyString._bytes;
                 return;
             }
 
             Debug.Assert(bytes.Length == ValueKeccak.MemorySize);
-            Bytes = Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(bytes));
+            _bytes = Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(bytes));
```

Comment:
Getting a bit confused here, to clarify, this do copy the bytes right? Just in an optimized way.

Reply:
Yes; or rather reinterprets to a `Vector256<byte>` and then assigns to the `Vector256<byte> _bytes` field.

Is a 256 bit / 32 byte copy as a single instruction